lib/include/openturns/OTtestcode.hxx

```diff
@@ -454,6 +429,596 @@ void assert_equal(const T & a, const T & b, const String errMsg = "")
+      throw TestFailed(OSS() << "operator==(copy) failed for " << distribution_);
+    if (distribution2 != distribution_)
+      throw TestFailed(OSS() << "operator!=(copy) failed for " << distribution_);
+    const Dirac dirac(42.0);
+    if (distribution_ == dirac)
+      throw TestFailed(OSS() << "operator==(other) failed for " << distribution_);
+    if (!(distribution2 != dirac))
+      throw TestFailed(OSS() << "operator!=(other) failed for " << distribution_);
+  }
+  
+  void checkPDF() const
+  {
+    LOGTRACE(OSS() << "checking PDF...");
+    const Sample sample(distribution_.getSample(10));
+    for (UnsignedInteger i = 0; i < sample.getSize(); ++ i)
+    {
+      const Point x(sample[i]);
+      const Scalar pdf = distribution_.computePDF(x);
+      LOGTRACE(OSS() << "pdf=" << pdf);
+      if (!(pdf > 0.0))
+        throw TestFailed(OSS() << "pdf(x) failed for " << distribution_);
+    }
+  }
+
+  void checkLogPDF() const
+  {
+    LOGTRACE(OSS() << "checking LogPDF...");
+    const Sample sample(distribution_.getSample(10));
+    for (UnsignedInteger i = 0; i < sample.getSize(); ++ i)
+    {
+      const Point x(sample[i]);
+      const Scalar lpdf2 = std::log(distribution_.computePDF(x));
+      LOGTRACE(OSS() << "log(pdf)=" << lpdf2);
+      const Scalar lpdf1 = distribution_.computeLogPDF(x);
+      LOGTRACE(OSS() << "logpdf  =" << lpdf1);
+      assert_almost_equal(lpdf1, lpdf2, 1e-5, 1e-8, distribution_.__repr__());
+    }
+  }
+
+  void checkCDF() const
+  {
+    LOGTRACE(OSS() << "checking CDF...");
+    // check CDF at bounds
+    const UnsignedInteger dimension = distribution_.getDimension();
+    const Point epsilon(dimension, std::pow(ResourceMap::GetAsScalar("Distribution-DefaultCDFEpsilon"), 1.0 / 3.0));
+    const Scalar cdflb = distribution_.computeCDF(distribution_.getRange().getLowerBound() - epsilon);
+    LOGTRACE(OSS() << "cdf(lb)=" << cdflb);
+    assert_almost_equal(cdflb, 0.0, 1e-5, 1e-5,  "cdf(lb) " + distribution_.__repr__());
+    const Scalar cdfub = distribution_.computeCDF(distribution_.getRange().getUpperBound() + epsilon);
+    LOGTRACE(OSS() << "cdf(ub)=" << cdfub);
+    assert_almost_equal(cdfub, 1.0, 1e-5, 1e-5,  "cdf(ub) " + distribution_.__repr__());
+
+    if (distribution_.isContinuous() && (dimension == 1))
+    {
+      const Sample sample(distribution_.getSample(5));
+      for (UnsignedInteger i = 0; i < sample.getSize(); ++ i)
+      {
+        const Point x(sample[i]);
+        const Scalar pdf1 = distribution_.computePDF(x);
+        LOGTRACE(OSS() << "pdf    =" << pdf1);
+        const Scalar cdf1 = distribution_.computeCDF(Point({x[0] + epsilon[0]}));
+        const Scalar cdf2 = distribution_.computeCDF(Point({x[0] - epsilon[0]}));
+        const Scalar pdf2 = (cdf1 - cdf2) / (2.0 * epsilon[0]);
+        LOGTRACE(OSS() << "pdf(fd)=" << pdf2);
+        assert_almost_equal(pdf1, pdf2, 1e-3, 1e-3,  "cdf " + distribution_.__repr__());
+      }
+    }
+  }
```

Comment:
At this stage, the CDF is consistent with the PDF. Another way to check the CDF is to get a sample and estimate the probability $\mathbb{P}(X \leq x)$ for some value of $x$. This can alternatively makes a check for `getSample()`. @jschueller : Should this be done?

Reply:
I think thats already covered by the kolmogorov fitting test